File: ui/src/api/system.ts
/**
 * System API
 */

import { invoke } from '@tauri-apps/api/core';
import type {
  CommandResponse,
  DoctorResult,
  ProviderStatus,
  SystemInfo,
} from '../types';

export async function runDoctor(): Promise<DoctorResult> {
  const response = await invoke<CommandResponse<DoctorResult>>('run_doctor');
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to run doctor');
  }
  return response.data;
}

export async function runProjectDoctor(projectPath: string): Promise<DoctorResult> {
  const response = await invoke<CommandResponse<DoctorResult>>('run_project_doctor', {
    projectPath,
  });
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to run project doctor');
  }
  return response.data;
}

export async function getSystemInfo(): Promise<SystemInfo> {
  const response = await invoke<CommandResponse<SystemInfo>>('get_system_info');
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to get system info');
  }
  return response.data;
}

export async function getProviderStatus(provider: string): Promise<ProviderStatus> {
  const response = await invoke<CommandResponse<ProviderStatus>>('get_provider_status', {
    provider,
  });
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to get provider status');
  }
  return response.data;
}

export async function getAllProviders(): Promise<Record<string, ProviderStatus>> {
  const response = await invoke<CommandResponse<Record<string, ProviderStatus>>>(
    'get_all_providers'
  );
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to get providers');
  }
  return response.data;
}

export async function checkUpdates(): Promise<{
  current_version: string;
  latest_version: string;
  update_available: boolean;
}> {
  const response = await invoke<
    CommandResponse<{
      current_version: string;
      latest_version: string;
      update_available: boolean;
    }>
  >('check_updates');
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to check updates');
  }
  return response.data;
}

export async function getVersion(): Promise<{ devflow: string; python: string }> {
  const response = await invoke<
    CommandResponse<{ devflow: string; python: string }>
  >('get_version');
  if (!response.success || !response.data) {
    throw new Error(response.error || 'Failed to get version');
  }
  return response.data;
}

File: ui/src-tauri/src/commands/system.rs
use super::{bridge_call, CommandResponse};
use crate::bridge::BridgeManager;
use serde::{Deserialize, Serialize};
use serde_json::{json, Value};
use std::sync::Arc;
use tauri::State;

#[derive(Debug, Serialize, Deserialize)]
pub struct DoctorResult {
    pub overall_status: String, // "healthy" | "warning" | "error"
    pub checks: Vec<DoctorCheck>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct DoctorCheck {
    pub name: String,
    pub category: String, // "tool" | "auth" | "config" | "infrastructure"
    pub status: String,   // "ok" | "warning" | "error" | "skipped"
    pub message: String,
    pub details: Option<Value>,
    pub fix_hint: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ProviderStatus {
    pub name: String,
    pub binary: String,
    pub available: bool,
    pub authenticated: bool,
    pub version: Option<String>,
    pub path: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SystemInfo {
    pub platform: String,
    pub os_version: String,
    pub devflow_version: String,
    pub python_version: String,
    pub docker_version: Option<String>,
    pub home_dir: String,
    pub config_dir: String,
}

/// Run system doctor
#[tauri::command]
pub fn run_doctor(bridge: State<Arc<BridgeManager>>) -> CommandResponse<Value> {
    match bridge_call(&bridge, "system.doctor", None) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Run doctor for specific project
#[tauri::command]
pub fn run_project_doctor(
    bridge: State<Arc<BridgeManager>>,
    project_path: String,
) -> CommandResponse<Value> {
    match bridge_call(
        &bridge,
        "system.doctor",
        Some(json!({ "path": project_path })),
    ) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Get system information
#[tauri::command]
pub fn get_system_info(bridge: State<Arc<BridgeManager>>) -> CommandResponse<Value> {
    match bridge_call(&bridge, "system.info", None) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Get provider status
#[tauri::command]
pub fn get_provider_status(
    bridge: State<Arc<BridgeManager>>,
    provider: String,
) -> CommandResponse<Value> {
    match bridge_call(
        &bridge,
        "system.provider_status",
        Some(json!({ "provider": provider })),
    ) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Get all provider statuses
#[tauri::command]
pub fn get_all_providers(bridge: State<Arc<BridgeManager>>) -> CommandResponse<Value> {
    match bridge_call(&bridge, "system.all_providers", None) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Check for updates
#[tauri::command]
pub fn check_updates(bridge: State<Arc<BridgeManager>>) -> CommandResponse<Value> {
    match bridge_call(&bridge, "system.check_updates", None) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}

/// Get DevFlow version
#[tauri::command]
pub fn get_version(bridge: State<Arc<BridgeManager>>) -> CommandResponse<Value> {
    match bridge_call(&bridge, "system.version", None) {
        Ok(data) => CommandResponse::ok(data),
        Err(e) => CommandResponse::err(e),
    }
}
